src/lib/IdeaService.ts

```diff
@@ -0,0 +1,296 @@
+      targetAudienceEvaluation
+    )
+
+    await this.saveIdea(idea)
+  }
+
+  async evaluateContentAndLongTermStrategy(id: string): Promise<void> {
+    if (!this.ideaAIService) {
+      throw new Error('IdeaAIService is not defined')
+    }
+
+    const idea = await this.repository.findById(id)
+
+    if (!idea) {
+      throw new Error(`Idea ${id} does not exist`)
+    }
+
+    const evaluation =
+      await this.ideaAIService.evaluateContentAndLongTermStrategy(
+        idea.getInitialProblem(),
+        idea.getInitialTargetAudience()
+      )
+
+    const contentAndStrategyEvaluation =
+      ContentAndLongTermStrategyEvaluation.New(
+        evaluation.valueProposition,
+        evaluation.mvpRecommendation,
+        evaluation.twoWeekPlan,
+        evaluation.mvpCostAndTimeline
+      )
+
+    idea.evaluateContentAndLongTermStrategy(contentAndStrategyEvaluation)
+
+    await this.saveIdea(idea)
+  }
+
+  async evaluateUserAcquisitionAndCompetitorAnalysis(
+    id: string
+  ): Promise<void> {
+    if (!this.ideaAIService) {
+      throw new Error('IdeaAIService is not defined')
+    }
+
+    const idea = await this.repository.findById(id)
+
+    if (!idea) {
+      throw new Error(`Idea ${id} does not exist`)
+    }
+
+    const evaluation =
+      await this.ideaAIService.evaluateUserAcquisitionAndCompetitorAnalysis(
+        idea.getInitialProblem(),
+        idea.getInitialTargetAudience()
+      )
+
+    const userAcquisitionAndCompetitorAnalysis =
+      UserAcquisitionAndCompetitorAnalysis.New(
+        evaluation.earlyAdoptersAcquisitionIdeas,
+        evaluation.competitorOverview,
+        evaluation.potentialProductNames,
+        evaluation.collaborationOpportunities
+      )
+
+    idea.evaluateUserAcquisitionAndCompetitorAnalysis(
+      userAcquisitionAndCompetitorAnalysis
+    )
+
+    await this.saveIdea(idea)
+  }
+
+  async evaluateContentStrategyAndGrowthPlan(id: string): Promise<void> {
+    if (!this.ideaAIService) {
+      throw new Error('IdeaAIService is not defined')
+    }
+
+    const idea = await this.repository.findById(id)
+
+    if (!idea) {
+      throw new Error(`Idea ${id} does not exist`)
+    }
+
+    const evaluation =
+      await this.ideaAIService.evaluateContentStrategyAndGrowthPlan(
+        idea.getInitialProblem(),
+        idea.getInitialTargetAudience()
+      )
+
+    const contentStrategyAndGrowthPlan = ContentStrategyAndGrowthPlan.New(
+      evaluation.contentMarketingIdeas,
+      evaluation.keyMetricsToTrackPostLaunch,
+      evaluation.recommendedToolsAndServices,
+      evaluation.caseStudyOutline
+    )
+
+    idea.evaluateContentStrategyAndGrowthPlan(contentStrategyAndGrowthPlan)
+
+    await this.saveIdea(idea)
+  }
+
+  async saveIdea(idea: Idea): Promise<void> {
+    await this.repository.save(idea)
+
+    await this.handleDomainEvents(idea)
+  }
+
+  private async handleDomainEvents(idea: Idea): Promise<void> {
+    for (const event of idea.getDomainEvents()) {
```

Comment:
Dispatch events through events service (pub/sub), instead of handling it manually.